hyperdome/server/hyperdome_server.py

```diff
@@ -59,25 +59,26 @@ def __init__(self, onion, local_only=False, shutdown_timeout=0):
         # init timing thread
         self.shutdown_timer = None
 
-    def choose_port(self):
+    @property
+    def port(self):
         """
         Choose a random port.
         """
-        self.port = get_available_port(17600, 17650)
+        if self._port is None:
+            self._port = get_available_port(17600, 17650)
+
+        return self._port
```

Comment:
I like this

Reply:
Yeah way better than having a get_port and a test to see if port is bound everywhere it's called.